tests/PHPStan/Analyser/data/non-empty-string.php

```diff
@@ -359,6 +360,10 @@ public function doFoo(string $s, string $nonEmpty, int $i, bool $bool, $constUni
 		assertType('int<0, max>', strlen($s));
 		assertType('int<1, max>', strlen($nonEmpty));
 		assertType('int<1, 2>', strlen($constUnion));
+		assertType('int<0, 4>', strlen($constUnionMixed));
+		assertType('3', strlen(123));
+		assertType('1', strlen(true));
+		assertType('0', strlen(false));
```

Comment:
what about `float`?
https://3v4l.org/XVRAE

maybe even `null`?
https://3v4l.org/J1pbq

Reply:
Good catch!
I will add support for these